### PR TITLE
New badge preview

### PR DESCRIPTION
**Pull request by FedeMadoery**
- [x] New badge preview
- [x] The badge logo with fill color options

### CODE REVIEW

**Comment by ramabit** [comment]
Screenshot 2022-12-06 at 12 23 38
With the new badge preview v2 component, it does not look good in the carousel

**Comment by ramabit** [comment]
Is the idea to have both versions of badge preview component available to be used? If so, then the badgePreviewV2 is needed in the index.ts

File: src/components/logos/LogoTheBadgeWithText/LogoTheBadgeWithText.tsx [hunk 1]
```diff
@@ -10,34 +10,37 @@ export const LogoTheBadgeWithText = (props: LogoProps) => {
...
     >
-      <path d="M272.012 81.3943V25.6899H254.568V12.4719H303.601V25.6899H286.157V81.3943H272.012Z" fill="white" />
+      <path
+        d="M272.012 81.3943V25.6899H254.568V12.4719H303.601V25.6899H286.157V81.3943H272.012Z"
+        fill={props.fill ?? 'white'}
+      />
       <path
         d="M313.559 81.3943V12.4719H327.704V38.9079H343.545V12.4719H357.689V81.3943H343.545V52.1259H327.704V81.3943H313.559Z"
-        fill="white"
+        fill={props.fill ?? 'white'}
```
Comment (ramabit) [inline src/components/logos/LogoTheBadgeWithText/LogoTheBadgeWithText.tsx:19]: please add a sample in storybook changing the fill color

File: src/components/molecules/BadgePreviewV2/BadgePreview.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      control: {
+        type: 'select', // Type 'select' is automatically inferred when 'options' is defined
+        labels: {
+          // 'labels' maps option values to string labels
+          backgroundOne: 'One',
+          backgroundTwo: 'Two',
+          backgroundThree: 'Three',
+          backgroundFour: 'Four',
+          backgroundFive: 'Five',
+          backgroundSix: 'Six',
```
Comment (ramabit) [inline src/components/molecules/BadgePreviewV2/BadgePreview.stories.tsx:37]: I guess in the future we should think about fancy names for each background, but this is ok for now

**Review by Agupane — CHANGES_REQUESTED** [review]
Aside from the code-comments, I tested it using storybook and I found some issues:

On "Badge Preview Long Texts" there is no default animattionsEffect object, so if you want to activate the animation on hover and you press "set object" you will get this error:

image

And even if I copy the animationsEffect from the "Badge preview with animation on hover" it does not work as the code already throwed the exception:

image

Aside from that, the other attributes seems to work fine, I updated everyone and it got automatically updated on the preview

Another thing, I'm not 100% sure it is intended to have it on this PR it will come later, but at least have it in mind:

The carousel preview does not rotate in any of the versions (badgev1 and badgev2), neither works the "selfRotate" property (or its too slow in time as default and I could not see the rotation, but I waited for around 1 minute and did not happen)

Another weird thing is that if I enable the selfRotation it won't rotate on the preview, but once I change the tab, start doing something (like writing this comment) and go back again, it will automatically do all the rotations together, which was supposed to happen earlier. So I'm not 100% because I didn't check the rotation code but seems that it could be an issue with some listener/event that's not being triggered properly and once the page refresh it does.

I will send you a video on discord about that. On the video, you saw that I could not make it rotate manually, but if I change the tab it will do it.

Aside from the issues, it the new badges looks amazing, great job!

**Review by Agupane — APPROVED** [review]
Now looks good

- Issues with the badge preview fixed
- The rotation on the carousel seems to be partially fixed, now it works better, the only thing missing is to fix the angle of the corners to make them easy to see and the rotation events after I change the tab, but we can proceed with this ticket and leave those issues for [this ticket](https://github.com/thebadge/thebadge-ui-library/issues/7).

**Pull request merged**